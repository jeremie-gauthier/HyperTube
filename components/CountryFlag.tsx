import React from "react";
import UKFlag from "../public/icons/united-kingdom.svg";
import FRFlag from "../public/icons/france.svg";
import ESFlag from "../public/icons/spain.svg";
import JAFlag from "../public/icons/japan.svg";

type CountryFlagProps = {
  lang: string;
};

export default function CountryFlag({
  lang,
  ...rest
}: CountryFlagProps): JSX.Element | null {
  const isUK = lang === "en";
  const isFrance = lang === "fr";
  const isSpain = lang === "es";
  const isJapan = lang === "ja";

  if (isUK) return <UKFlag {...rest} />;
  if (isFrance) return <FRFlag {...rest} />;
  if (isSpain) return <ESFlag {...rest} />;
  if (isJapan) return <JAFlag {...rest} />;

  return null;
}
