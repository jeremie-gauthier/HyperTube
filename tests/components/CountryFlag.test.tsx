import React from "react";
import CountryFlag from "@/components/CountryFlag";
import renderer from "react-test-renderer";

test("<CountryFlag /> should render the French Flag", () => {
  const tree = renderer.create(<CountryFlag lang="fr" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("<CountryFlag /> should render null", () => {
  const tree = renderer.create(<CountryFlag lang="wrong_value" />).toJSON();
  expect(tree).toMatchSnapshot();
});
