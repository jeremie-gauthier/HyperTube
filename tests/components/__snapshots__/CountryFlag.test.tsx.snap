// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountryFlag /> should render null 1`] = `null`;

exports[`<CountryFlag /> should render the French Flag 1`] = `
<svg
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M0 0h170.67v512H0z"
    fill="#41479b"
  />
  <path
    d="M170.67 0h170.67v512H170.67z"
    fill="#f5f5f5"
  />
  <path
    d="M341.33 0H512v512H341.33z"
    fill="#ff4b55"
  />
</svg>
`;

exports[`<LangSettings /> should render the list of languages when clicked 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-hidden="true"
        class="frame isTouchScreen"
        data-testid="lang-flag-clickable"
        role="button"
      >
        <svg
          class="countryFlag"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#41479b"
          >
            <path
              d="M188.632 0H0v39.954L188.632 163.54zM0 136.598v52.034h79.419zM0 323.369v52.033l79.419-52.033zM512 39.954V0H323.368v163.54zM512 375.402v-52.033h-79.419zM323.368 348.46V512H512v-39.954zM512 188.632v-52.034l-79.419 52.034zM0 472.046V512h188.632V348.46z"
            />
          </g>
          <g
            fill="#f5f5f5"
          >
            <path
              d="M512 472.046v-39.928L344.465 323.369h49.496L512 399.989v-24.587l-79.419-52.033H512v-26.948H296.421V512h26.947V348.46zM0 296.421v26.948h79.419L0 375.402v37.801l138.395-89.834h49.496L0 445.332v26.714L188.632 348.46V512h26.947V296.421zM215.579 0h-26.947v163.54L0 39.954v43.725l161.684 104.953h-49.496L0 115.807v20.791l79.419 52.034H0v26.947h215.579zM512 215.579v-26.947h-79.419L512 136.598V98.314l-139.136 90.318h-49.496L512 66.185V39.954L323.368 163.54V0h-26.947v215.579z"
            />
          </g>
          <g
            fill="#ff4b55"
          >
            <path
              d="M512 296.421v-80.842H296.421V0h-80.842v215.579H0v80.842h215.579V512h80.842V296.421z"
            />
            <path
              d="M138.395 323.369L0 413.203v32.129l187.891-121.963zM344.465 323.369L512 432.118v-32.129l-118.039-76.62zM161.684 188.632L0 83.679v32.128l112.188 72.825zM372.864 188.632L512 98.314V66.185L323.368 188.632z"
            />
          </g>
        </svg>
        <div
          class="floaterMobile"
        >
          <div
            class="flexCol floaterContent"
          >
            <button
              type="button"
            >
              <svg
                height="15"
                viewBox="0 0 512 512"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="#41479b"
                >
                  <path
                    d="M188.632 0H0v39.954L188.632 163.54zM0 136.598v52.034h79.419zM0 323.369v52.033l79.419-52.033zM512 39.954V0H323.368v163.54zM512 375.402v-52.033h-79.419zM323.368 348.46V512H512v-39.954zM512 188.632v-52.034l-79.419 52.034zM0 472.046V512h188.632V348.46z"
                  />
                </g>
                <g
                  fill="#f5f5f5"
                >
                  <path
                    d="M512 472.046v-39.928L344.465 323.369h49.496L512 399.989v-24.587l-79.419-52.033H512v-26.948H296.421V512h26.947V348.46zM0 296.421v26.948h79.419L0 375.402v37.801l138.395-89.834h49.496L0 445.332v26.714L188.632 348.46V512h26.947V296.421zM215.579 0h-26.947v163.54L0 39.954v43.725l161.684 104.953h-49.496L0 115.807v20.791l79.419 52.034H0v26.947h215.579zM512 215.579v-26.947h-79.419L512 136.598V98.314l-139.136 90.318h-49.496L512 66.185V39.954L323.368 163.54V0h-26.947v215.579z"
                  />
                </g>
                <g
                  fill="#ff4b55"
                >
                  <path
                    d="M512 296.421v-80.842H296.421V0h-80.842v215.579H0v80.842h215.579V512h80.842V296.421z"
                  />
                  <path
                    d="M138.395 323.369L0 413.203v32.129l187.891-121.963zM344.465 323.369L512 432.118v-32.129l-118.039-76.62zM161.684 188.632L0 83.679v32.128l112.188 72.825zM372.864 188.632L512 98.314V66.185L323.368 188.632z"
                  />
                </g>
              </svg>
              <span>
                common.lang.english
              </span>
            </button>
            <button
              type="button"
            >
              <svg
                height="15"
                viewBox="0 0 512 512"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h170.67v512H0z"
                  fill="#41479b"
                />
                <path
                  d="M170.67 0h170.67v512H170.67z"
                  fill="#f5f5f5"
                />
                <path
                  d="M341.33 0H512v512H341.33z"
                  fill="#ff4b55"
                />
              </svg>
              <span>
                common.lang.french
              </span>
            </button>
            <button
              type="button"
            >
              <svg
                height="15"
                viewBox="0 0 512 512"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1.985 0h508.03v512H1.985z"
                  fill="#c8414b"
                />
                <path
                  d="M1.985 107.79h508.03v296.42H1.985z"
                  fill="#ffd250"
                />
                <path
                  d="M223.347 256.409l8.191-33.404c.754-3.076-1.829-5.994-5.306-5.994h-5.77c-3.477 0-6.061 2.918-5.306 5.994l8.191 33.404z"
                  fill="#c8414b"
                />
                <path
                  d="M213.681 238.89h19.322v74.42h-19.322z"
                  fill="#f5f5f5"
                />
                <path
                  d="M208.861 230.14h28.984v8.756h-28.984z"
                  fill="#fab446"
                />
                <g
                  fill="#c8414b"
                >
                  <path
                    d="M189.531 256.41h48.3v8.756h-48.3zM237.839 291.429l-28.983-8.755v-8.755l28.983 8.755zM78.435 256.409l8.191-33.404c.754-3.076-1.829-5.994-5.306-5.994h-5.77c-3.477 0-6.061 2.918-5.306 5.994l8.191 33.404z"
                  />
                </g>
                <path
                  d="M112.248 230.143c-5.335 0-9.661 3.919-9.661 8.756v56.908c0 10.638 10.955 30.643 48.305 30.643s48.305-20.006 48.305-30.643v-56.908c0-4.835-4.325-8.756-9.661-8.756h-77.288z"
                  fill="#f5f5f5"
                />
                <g
                  fill="#c8414b"
                >
                  <path
                    d="M150.891 273.919h-48.305V239.34a9.197 9.197 0 0 1 9.197-9.197h39.107v43.776h.001zM150.891 273.919h48.305v20.784c0 12.698-10.294 22.992-22.992 22.992h-2.32c-12.698 0-22.992-10.294-22.992-22.992l-.001-20.784z"
                  />
                </g>
                <path
                  d="M102.587 273.919h48.305v20.784c0 12.698-10.294 22.992-22.992 22.992h-2.32c-12.698 0-22.992-10.294-22.992-22.992l-.001-20.784z"
                  fill="#fab446"
                />
                <g
                  fill="#c8414b"
                >
                  <path
                    d="M141.231 313.218v-39.299h-9.661v43.334c3.592-.661 6.84-2.103 9.661-4.035zM121.909 317.253v-43.334h-9.661v39.299c2.821 1.932 6.068 3.374 9.661 4.035z"
                  />
                </g>
                <path
                  d="M112.241 256.41h28.984v8.756h-28.984z"
                  fill="#ffb441"
                />
                <g
                  fill="#fab446"
                >
                  <path
                    d="M112.241 238.89h28.984v8.756h-28.984z"
                  />
                  <path
                    d="M117.081 244.1h19.322v15.861h-19.322z"
                  />
                </g>
                <path
                  d="M68.774 238.89h19.322v74.42H68.774z"
                  fill="#f5f5f5"
                />
                <g
                  fill="#fab446"
                >
                  <path
                    d="M63.941 308.94h28.984v8.756H63.941zM63.941 230.14h28.984v8.756H63.941z"
                  />
                </g>
                <path
                  d="M59.111 317.7h38.643v8.756H59.111z"
                  fill="#5064aa"
                />
                <path
                  d="M213.681 308.94h28.984v8.756h-28.984z"
                  fill="#fab446"
                />
                <path
                  d="M204.031 317.7h38.643v8.756h-38.643z"
                  fill="#5064aa"
                />
                <path
                  d="M121.911 221.39h57.967v8.756h-57.967z"
                  fill="#fab446"
                />
                <path
                  d="M146.061 195.13h9.661v26.27h-9.661z"
                  fill="#ffb441"
                />
                <g
                  fill="#f5f5f5"
                >
                  <path
                    d="M141.231 208.255c-7.991 0-14.491-5.891-14.491-13.132s6.5-13.132 14.491-13.132 14.491 5.891 14.491 13.132c-.001 7.241-6.501 13.132-14.491 13.132zm0-17.51c-2.665 0-4.83 1.963-4.83 4.378s2.165 4.378 4.83 4.378 4.83-1.963 4.83-4.378c0-2.416-2.165-4.378-4.83-4.378z"
                  />
                  <path
                    d="M160.552 208.255c-7.991 0-14.491-5.891-14.491-13.132s6.5-13.132 14.491-13.132 14.491 5.891 14.491 13.132c-.001 7.241-6.5 13.132-14.491 13.132zm0-17.51c-2.665 0-4.83 1.963-4.83 4.378s2.165 4.378 4.83 4.378 4.83-1.963 4.83-4.378c0-2.416-2.165-4.378-4.83-4.378z"
                  />
                  <path
                    d="M179.874 217.011c-7.991 0-14.491-5.891-14.491-13.132s6.5-13.132 14.491-13.132 14.491 5.891 14.491 13.132-6.501 13.132-14.491 13.132zm0-17.511c-2.665 0-4.83 1.963-4.83 4.378s2.165 4.378 4.83 4.378 4.83-1.963 4.83-4.378c0-2.416-2.165-4.378-4.83-4.378zM121.909 217.011c-7.991 0-14.491-5.891-14.491-13.132s6.5-13.132 14.491-13.132 14.491 5.891 14.491 13.132c-.001 7.241-6.501 13.132-14.491 13.132zm0-17.511c-2.665 0-4.83 1.963-4.83 4.378s2.165 4.378 4.83 4.378 4.83-1.963 4.83-4.378c0-2.416-2.165-4.378-4.83-4.378z"
                  />
                </g>
                <path
                  d="M179.874 291.429v4.378c0 2.414-2.167 4.378-4.83 4.378s-4.83-1.964-4.83-4.378v-4.378h9.66m9.66-8.755h-28.983v13.132c0 7.241 6.501 13.132 14.491 13.132 7.991 0 14.491-5.891 14.491-13.132l.001-13.132z"
                  fill="#fab446"
                />
                <path
                  d="M175.507 265.163h-.928a9.197 9.197 0 0 1-9.197-9.197v-7.872a9.197 9.197 0 0 1 9.197-9.197h.928a9.197 9.197 0 0 1 9.197 9.197v7.872a9.196 9.196 0 0 1-9.197 9.197z"
                  fill="#ffa0d2"
                />
                <ellipse
                  cx="150.891"
                  cy="273.92"
                  fill="#5064aa"
                  rx="14.491"
                  ry="13.13"
                />
                <path
                  d="M146.061 177.61h9.661v26.27h-9.661z"
                  fill="#fab446"
                />
                <path
                  d="M121.909 221.388l-9.661-8.756 5.659-5.129c8.748-7.928 20.613-12.381 32.984-12.381 12.371 0 24.237 4.454 32.984 12.381l5.659 5.129-9.661 8.756h-57.964z"
                  fill="#c8414b"
                />
                <g
                  fill="#ffd250"
                >
                  <ellipse
                    cx="150.891"
                    cy="212.63"
                    rx="4.83"
                    ry="4.378"
                  />
                  <ellipse
                    cx="131.571"
                    cy="212.63"
                    rx="4.83"
                    ry="4.378"
                  />
                  <ellipse
                    cx="170.211"
                    cy="212.63"
                    rx="4.83"
                    ry="4.378"
                  />
                </g>
                <g
                  fill="#c8414b"
                >
                  <path
                    d="M63.941 256.41h48.3v8.756h-48.3zM63.943 291.429l28.983-8.755v-8.755l-28.983 8.755z"
                  />
                </g>
              </svg>
              <span>
                common.lang.spanish
              </span>
            </button>
            <button
              type="button"
            >
              <svg
                height="15"
                viewBox="0 0 512 512"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h512v512H0z"
                  fill="#f5f5f5"
                />
                <circle
                  cx="256"
                  cy="256"
                  fill="#ff4b55"
                  r="97.1"
                />
              </svg>
              <span>
                common.lang.japanese
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="frame isNotTouchScreen"
        data-testid="lang-flag-hoverable"
      >
        <svg
          class="countryFlag"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#41479b"
          >
            <path
              d="M188.632 0H0v39.954L188.632 163.54zM0 136.598v52.034h79.419zM0 323.369v52.033l79.419-52.033zM512 39.954V0H323.368v163.54zM512 375.402v-52.033h-79.419zM323.368 348.46V512H512v-39.954zM512 188.632v-52.034l-79.419 52.034zM0 472.046V512h188.632V348.46z"
            />
          </g>
          <g
            fill="#f5f5f5"
          >
            <path
              d="M512 472.046v-39.928L344.465 323.369h49.496L512 399.989v-24.587l-79.419-52.033H512v-26.948H296.421V512h26.947V348.46zM0 296.421v26.948h79.419L0 375.402v37.801l138.395-89.834h49.496L0 445.332v26.714L188.632 348.46V512h26.947V296.421zM215.579 0h-26.947v163.54L0 39.954v43.725l161.684 104.953h-49.496L0 115.807v20.791l79.419 52.034H0v26.947h215.579zM512 215.579v-26.947h-79.419L512 136.598V98.314l-139.136 90.318h-49.496L512 66.185V39.954L323.368 163.54V0h-26.947v215.579z"
            />
          </g>
          <g
            fill="#ff4b55"
          >
            <path
              d="M512 296.421v-80.842H296.421V0h-80.842v215.579H0v80.842h215.579V512h80.842V296.421z"
            />
            <path
              d="M138.395 323.369L0 413.203v32.129l187.891-121.963zM344.465 323.369L512 432.118v-32.129l-118.039-76.62zM161.684 188.632L0 83.679v32.128l112.188 72.825zM372.864 188.632L512 98.314V66.185L323.368 188.632z"
            />
          </g>
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-hidden="true"
      class="frame isTouchScreen"
      data-testid="lang-flag-clickable"
      role="button"
    >
      <svg
        class="countryFlag"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="#41479b"
        >
          <path
            d="M188.632 0H0v39.954L188.632 163.54zM0 136.598v52.034h79.419zM0 323.369v52.033l79.419-52.033zM512 39.954V0H323.368v163.54zM512 375.402v-52.033h-79.419zM323.368 348.46V512H512v-39.954zM512 188.632v-52.034l-79.419 52.034zM0 472.046V512h188.632V348.46z"
          />
        </g>
        <g
          fill="#f5f5f5"
        >
          <path
            d="M512 472.046v-39.928L344.465 323.369h49.496L512 399.989v-24.587l-79.419-52.033H512v-26.948H296.421V512h26.947V348.46zM0 296.421v26.948h79.419L0 375.402v37.801l138.395-89.834h49.496L0 445.332v26.714L188.632 348.46V512h26.947V296.421zM215.579 0h-26.947v163.54L0 39.954v43.725l161.684 104.953h-49.496L0 115.807v20.791l79.419 52.034H0v26.947h215.579zM512 215.579v-26.947h-79.419L512 136.598V98.314l-139.136 90.318h-49.496L512 66.185V39.954L323.368 163.54V0h-26.947v215.579z"
          />
        </g>
        <g
          fill="#ff4b55"
        >
          <path
            d="M512 296.421v-80.842H296.421V0h-80.842v215.579H0v80.842h215.579V512h80.842V296.421z"
          />
          <path
            d="M138.395 323.369L0 413.203v32.129l187.891-121.963zM344.465 323.369L512 432.118v-32.129l-118.039-76.62zM161.684 188.632L0 83.679v32.128l112.188 72.825zM372.864 188.632L512 98.314V66.185L323.368 188.632z"
          />
        </g>
      </svg>
      <div
        class="floaterMobile"
      >
        <div
          class="flexCol floaterContent"
        >
          <button
            type="button"
          >
            <svg
              height="15"
              viewBox="0 0 512 512"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="#41479b"
              >
                <path
                  d="M188.632 0H0v39.954L188.632 163.54zM0 136.598v52.034h79.419zM0 323.369v52.033l79.419-52.033zM512 39.954V0H323.368v163.54zM512 375.402v-52.033h-79.419zM323.368 348.46V512H512v-39.954zM512 188.632v-52.034l-79.419 52.034zM0 472.046V512h188.632V348.46z"
                />
              </g>
              <g
                fill="#f5f5f5"
              >
                <path
                  d="M512 472.046v-39.928L344.465 323.369h49.496L512 399.989v-24.587l-79.419-52.033H512v-26.948H296.421V512h26.947V348.46zM0 296.421v26.948h79.419L0 375.402v37.801l138.395-89.834h49.496L0 445.332v26.714L188.632 348.46V512h26.947V296.421zM215.579 0h-26.947v163.54L0 39.954v43.725l161.684 104.953h-49.496L0 115.807v20.791l79.419 52.034H0v26.947h215.579zM512 215.579v-26.947h-79.419L512 136.598V98.314l-139.136 90.318h-49.496L512 66.185V39.954L323.368 163.54V0h-26.947v215.579z"
                />
              </g>
              <g
                fill="#ff4b55"
              >
                <path
                  d="M512 296.421v-80.842H296.421V0h-80.842v215.579H0v80.842h215.579V512h80.842V296.421z"
                />
                <path
                  d="M138.395 323.369L0 413.203v32.129l187.891-121.963zM344.465 323.369L512 432.118v-32.129l-118.039-76.62zM161.684 188.632L0 83.679v32.128l112.188 72.825zM372.864 188.632L512 98.314V66.185L323.368 188.632z"
                />
              </g>
            </svg>
            <span>
              common.lang.english
            </span>
          </button>
          <button
            type="button"
          >
            <svg
              height="15"
              viewBox="0 0 512 512"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h170.67v512H0z"
                fill="#41479b"
              />
              <path
                d="M170.67 0h170.67v512H170.67z"
                fill="#f5f5f5"
              />
              <path
                d="M341.33 0H512v512H341.33z"
                fill="#ff4b55"
              />
            </svg>
            <span>
              common.lang.french
            </span>
          </button>
          <button
            type="button"
          >
            <svg
              height="15"
              viewBox="0 0 512 512"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1.985 0h508.03v512H1.985z"
                fill="#c8414b"
              />
              <path
                d="M1.985 107.79h508.03v296.42H1.985z"
                fill="#ffd250"
              />
              <path
                d="M223.347 256.409l8.191-33.404c.754-3.076-1.829-5.994-5.306-5.994h-5.77c-3.477 0-6.061 2.918-5.306 5.994l8.191 33.404z"
                fill="#c8414b"
              />
              <path
                d="M213.681 238.89h19.322v74.42h-19.322z"
                fill="#f5f5f5"
              />
              <path
                d="M208.861 230.14h28.984v8.756h-28.984z"
                fill="#fab446"
              />
              <g
                fill="#c8414b"
              >
                <path
                  d="M189.531 256.41h48.3v8.756h-48.3zM237.839 291.429l-28.983-8.755v-8.755l28.983 8.755zM78.435 256.409l8.191-33.404c.754-3.076-1.829-5.994-5.306-5.994h-5.77c-3.477 0-6.061 2.918-5.306 5.994l8.191 33.404z"
                />
              </g>
              <path
                d="M112.248 230.143c-5.335 0-9.661 3.919-9.661 8.756v56.908c0 10.638 10.955 30.643 48.305 30.643s48.305-20.006 48.305-30.643v-56.908c0-4.835-4.325-8.756-9.661-8.756h-77.288z"
                fill="#f5f5f5"
              />
              <g
                fill="#c8414b"
              >
                <path
                  d="M150.891 273.919h-48.305V239.34a9.197 9.197 0 0 1 9.197-9.197h39.107v43.776h.001zM150.891 273.919h48.305v20.784c0 12.698-10.294 22.992-22.992 22.992h-2.32c-12.698 0-22.992-10.294-22.992-22.992l-.001-20.784z"
                />
              </g>
              <path
                d="M102.587 273.919h48.305v20.784c0 12.698-10.294 22.992-22.992 22.992h-2.32c-12.698 0-22.992-10.294-22.992-22.992l-.001-20.784z"
                fill="#fab446"
              />
              <g
                fill="#c8414b"
              >
                <path
                  d="M141.231 313.218v-39.299h-9.661v43.334c3.592-.661 6.84-2.103 9.661-4.035zM121.909 317.253v-43.334h-9.661v39.299c2.821 1.932 6.068 3.374 9.661 4.035z"
                />
              </g>
              <path
                d="M112.241 256.41h28.984v8.756h-28.984z"
                fill="#ffb441"
              />
              <g
                fill="#fab446"
              >
                <path
                  d="M112.241 238.89h28.984v8.756h-28.984z"
                />
                <path
                  d="M117.081 244.1h19.322v15.861h-19.322z"
                />
              </g>
              <path
                d="M68.774 238.89h19.322v74.42H68.774z"
                fill="#f5f5f5"
              />
              <g
                fill="#fab446"
              >
                <path
                  d="M63.941 308.94h28.984v8.756H63.941zM63.941 230.14h28.984v8.756H63.941z"
                />
              </g>
              <path
                d="M59.111 317.7h38.643v8.756H59.111z"
                fill="#5064aa"
              />
              <path
                d="M213.681 308.94h28.984v8.756h-28.984z"
                fill="#fab446"
              />
              <path
                d="M204.031 317.7h38.643v8.756h-38.643z"
                fill="#5064aa"
              />
              <path
                d="M121.911 221.39h57.967v8.756h-57.967z"
                fill="#fab446"
              />
              <path
                d="M146.061 195.13h9.661v26.27h-9.661z"
                fill="#ffb441"
              />
              <g
                fill="#f5f5f5"
              >
                <path
                  d="M141.231 208.255c-7.991 0-14.491-5.891-14.491-13.132s6.5-13.132 14.491-13.132 14.491 5.891 14.491 13.132c-.001 7.241-6.501 13.132-14.491 13.132zm0-17.51c-2.665 0-4.83 1.963-4.83 4.378s2.165 4.378 4.83 4.378 4.83-1.963 4.83-4.378c0-2.416-2.165-4.378-4.83-4.378z"
                />
                <path
                  d="M160.552 208.255c-7.991 0-14.491-5.891-14.491-13.132s6.5-13.132 14.491-13.132 14.491 5.891 14.491 13.132c-.001 7.241-6.5 13.132-14.491 13.132zm0-17.51c-2.665 0-4.83 1.963-4.83 4.378s2.165 4.378 4.83 4.378 4.83-1.963 4.83-4.378c0-2.416-2.165-4.378-4.83-4.378z"
                />
                <path
                  d="M179.874 217.011c-7.991 0-14.491-5.891-14.491-13.132s6.5-13.132 14.491-13.132 14.491 5.891 14.491 13.132-6.501 13.132-14.491 13.132zm0-17.511c-2.665 0-4.83 1.963-4.83 4.378s2.165 4.378 4.83 4.378 4.83-1.963 4.83-4.378c0-2.416-2.165-4.378-4.83-4.378zM121.909 217.011c-7.991 0-14.491-5.891-14.491-13.132s6.5-13.132 14.491-13.132 14.491 5.891 14.491 13.132c-.001 7.241-6.501 13.132-14.491 13.132zm0-17.511c-2.665 0-4.83 1.963-4.83 4.378s2.165 4.378 4.83 4.378 4.83-1.963 4.83-4.378c0-2.416-2.165-4.378-4.83-4.378z"
                />
              </g>
              <path
                d="M179.874 291.429v4.378c0 2.414-2.167 4.378-4.83 4.378s-4.83-1.964-4.83-4.378v-4.378h9.66m9.66-8.755h-28.983v13.132c0 7.241 6.501 13.132 14.491 13.132 7.991 0 14.491-5.891 14.491-13.132l.001-13.132z"
                fill="#fab446"
              />
              <path
                d="M175.507 265.163h-.928a9.197 9.197 0 0 1-9.197-9.197v-7.872a9.197 9.197 0 0 1 9.197-9.197h.928a9.197 9.197 0 0 1 9.197 9.197v7.872a9.196 9.196 0 0 1-9.197 9.197z"
                fill="#ffa0d2"
              />
              <ellipse
                cx="150.891"
                cy="273.92"
                fill="#5064aa"
                rx="14.491"
                ry="13.13"
              />
              <path
                d="M146.061 177.61h9.661v26.27h-9.661z"
                fill="#fab446"
              />
              <path
                d="M121.909 221.388l-9.661-8.756 5.659-5.129c8.748-7.928 20.613-12.381 32.984-12.381 12.371 0 24.237 4.454 32.984 12.381l5.659 5.129-9.661 8.756h-57.964z"
                fill="#c8414b"
              />
              <g
                fill="#ffd250"
              >
                <ellipse
                  cx="150.891"
                  cy="212.63"
                  rx="4.83"
                  ry="4.378"
                />
                <ellipse
                  cx="131.571"
                  cy="212.63"
                  rx="4.83"
                  ry="4.378"
                />
                <ellipse
                  cx="170.211"
                  cy="212.63"
                  rx="4.83"
                  ry="4.378"
                />
              </g>
              <g
                fill="#c8414b"
              >
                <path
                  d="M63.941 256.41h48.3v8.756h-48.3zM63.943 291.429l28.983-8.755v-8.755l-28.983 8.755z"
                />
              </g>
            </svg>
            <span>
              common.lang.spanish
            </span>
          </button>
          <button
            type="button"
          >
            <svg
              height="15"
              viewBox="0 0 512 512"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h512v512H0z"
                fill="#f5f5f5"
              />
              <circle
                cx="256"
                cy="256"
                fill="#ff4b55"
                r="97.1"
              />
            </svg>
            <span>
              common.lang.japanese
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="frame isNotTouchScreen"
      data-testid="lang-flag-hoverable"
    >
      <svg
        class="countryFlag"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="#41479b"
        >
          <path
            d="M188.632 0H0v39.954L188.632 163.54zM0 136.598v52.034h79.419zM0 323.369v52.033l79.419-52.033zM512 39.954V0H323.368v163.54zM512 375.402v-52.033h-79.419zM323.368 348.46V512H512v-39.954zM512 188.632v-52.034l-79.419 52.034zM0 472.046V512h188.632V348.46z"
          />
        </g>
        <g
          fill="#f5f5f5"
        >
          <path
            d="M512 472.046v-39.928L344.465 323.369h49.496L512 399.989v-24.587l-79.419-52.033H512v-26.948H296.421V512h26.947V348.46zM0 296.421v26.948h79.419L0 375.402v37.801l138.395-89.834h49.496L0 445.332v26.714L188.632 348.46V512h26.947V296.421zM215.579 0h-26.947v163.54L0 39.954v43.725l161.684 104.953h-49.496L0 115.807v20.791l79.419 52.034H0v26.947h215.579zM512 215.579v-26.947h-79.419L512 136.598V98.314l-139.136 90.318h-49.496L512 66.185V39.954L323.368 163.54V0h-26.947v215.579z"
          />
        </g>
        <g
          fill="#ff4b55"
        >
          <path
            d="M512 296.421v-80.842H296.421V0h-80.842v215.579H0v80.842h215.579V512h80.842V296.421z"
          />
          <path
            d="M138.395 323.369L0 413.203v32.129l187.891-121.963zM344.465 323.369L512 432.118v-32.129l-118.039-76.62zM161.684 188.632L0 83.679v32.128l112.188 72.825zM372.864 188.632L512 98.314V66.185L323.368 188.632z"
          />
        </g>
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
